ServiceApp/server/src/utils/credentialHelper.ts

```diff
@@ -107,10 +179,83 @@ export async function processReceivedCredentialForUser(unstructuredData: any) {
         console.log('Credential Verification Error: ', e);
     }
 }
+
+export async function createAuthenticationPresentationC2(): Promise<VerifiablePresentation> {
+    // 1.25 Sign DID Authentication
+    return new Promise<VerifiablePresentation>(async (resolve, reject) => {
+        try {
+            const challenge = Date.now().toString();
+            const did: any = await readData('didC2');
+
+            // Create a default client configuration from the parent config network.
+            const config = Config.fromNetwork(Network.mainnet());
+
+            // Create a client instance to publish messages to the Tangle.
+            const client = Client.fromConfig(config);
+
+            // Resolve a DID.
+            // Read DID Document might fail when no DID is actually located at the root - Unlikely as it is the DID of this instance
+            //@ts-ignore
+            const resolveRequest = await client.resolve(did.id);
+
+            const issuerDID = resolveRequest.document;
+
+            let credentialSubject = {
+                id: issuerDID.id.toString(),
+                challenge: challenge
+            };
+
+            const unsignedVc = VerifiableCredential.extend({
+                id: "http://example.edu/credentials/3732", //TODO: ???
```

Comment:
This is a pre-defined credential format. You don't need it here like this, just replace with some identifier specific to this project

Reply:
fixed in 3c6978c6f75aa8ed7718b8a8b6f9e08a561585a2, though Im not sure why there is both a field `type` and `id` required.